lib/Db/FeedMapperV2.php

```diff
@@ -158,20 +160,33 @@ public function findAllFromFolder(?int $id): array
      * @param string   $userId
      * @param int      $id
      * @param int|null $maxItemID
+     *
+     * @throws \OCP\DB\Exception
+     *
+     * @TODO Update for NC 21
      */
     public function read(string $userId, int $id, ?int $maxItemID = null): void
     {
-        $builder = $this->db->getQueryBuilder();
-        $builder->update(ItemMapperV2::TABLE_NAME, 'items')
+        $idBuilder = $this->db->getQueryBuilder();
+        $idBuilder->select('id')
+            ->from(ItemMapperV2::TABLE_NAME, 'items')
             ->innerJoin('items', FeedMapperV2::TABLE_NAME, 'feeds', 'items.feed_id = feeds.id')
-            ->setValue('unread', 0)
             ->andWhere('feeds.user_id = :userId')
             ->andWhere('feeds.id = :feedId')
             ->setParameter('userId', $userId)
             ->setParameter('feedId', $id);
 
+        $idList = $this->db->executeQuery($idBuilder->getSQL())
+                           ->fetch(PDO::FETCH_COLUMN);
```

Comment:
`getSQL()` does not contain the parameters, so they have to be include in `executeQuery()`
(same for `executeUpdate()` in L193, which I can't mark in this review)
```suggestion
            ->andWhere('feeds.id = :feedId');

        $idList = $this->db->executeQuery($idBuilder->getSQL(), ['userId' => $userId, 'feedId' => $id])
                           ->fetch(PDO::FETCH_COLUMN);
```
or
```suggestion
            ->andWhere('feeds.id = :feedId')
            ->setParameter('userId', $userId)
            ->setParameter('feedId', $id);

        $idList = $this->db->executeQuery($idBuilder->getSQL(), $idBuilder->getParameters())
                           ->fetch(PDO::FETCH_COLUMN);
```